projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.php

```diff
@@ -0,0 +1,196 @@
+ *
+ * @return void
+ */
+function wpcom_create_subscribe_template() {
+	if ( should_enable_subscriber_modal() ) {
+		$post = get_page_by_path( get_subscribe_template_slug(), OBJECT, 'wp_template_part' );
+
+		if ( ! $post ) {
+			$template = array(
+				'slug'         => get_subscribe_template_slug(),
+				'post_name'    => get_subscribe_template_slug(),
+				'post_title'   => get_subscribe_template_title(),
+				'post_content' => get_subscribe_template_content(),
+				'post_status'  => 'publish',
+				'post_author'  => 1,
+				'post_type'    => 'wp_template_part',
+				'scope'        => array(),
+				'tax_input'    => array(
+					'wp_theme' => get_option( 'stylesheet' ),
+				),
+			);
+			wp_insert_post( $template );
+		}
+	}
+}
+add_action( 'wp_loaded', 'wpcom_create_subscribe_template' );
+
+/**
+ * Adds modal with Subscribe Modal content.
+ *
+ * @return void
+ */
+function wpcom_add_subscribe_modal_to_frontend() {
+	if ( should_enable_subscriber_modal() && ! is_admin() ) {
+		$posts = get_posts(
+			array(
+				'post_type'   => 'wp_template_part',
+				'post_status' => 'publish',
+				'numberposts' => -1,
+			)
+		);
+
+		$subscribe_template = reset(
+			array_filter(
+				$posts,
+				function ( $post ) {
+					return $post->post_name === get_subscribe_template_slug();
+				}
+			)
+		);
+
+		$blocks = parse_blocks( $subscribe_template->post_content );
+
+		/*
+		 * Jetpack requires escaping html for the render_block() call below.
+		 * I'm using wp_kses() and specifying the allowed tags.
+		 * But this is fragile: users can edit the Subscribe Modal
+		 * template and add tags that are not allowed here.
+		 */
+		$allowed_html = array(
+			'div'    => array(
+				'class' => array(),
+				'style' => array(),
+			),
+			'h2'     => array(
+				'class' => array(),
+				'style' => array(),
+			),
+			'p'      => array(
+				'type'  => array(),
+				'class' => array(),
+				'style' => array(),
+				'id'    => array(),
+			),
+			'form'   => array(
+				'action'                 => array(),
+				'method'                 => array(),
+				'accept-charset'         => array(),
+				'data-blog'              => array(),
+				'data-post_access_level' => array(),
+				'id'                     => array(),
+			),
+			'label'  => array(
+				'id'    => array(),
+				'for'   => array(),
+				'class' => array(),
+			),
+			'input'  => array(
+				'id'          => array(),
+				'type'        => array(),
+				'name'        => array(),
+				'value'       => array(),
+				'class'       => array(),
+				'style'       => array(),
+				'placeholder' => array(),
+				'required'    => array(),
+			),
+			'button' => array(
+				'type'  => array(),
+				'class' => array(),
+				'style' => array(),
+			),
+		);
+
+		?>
+			<div id="wpcom-subscribe-modal">
+				<div class="modal-content">
+					<?php
+					foreach ( $blocks as $block ) {
+						echo wp_kses( render_block( $block ), $allowed_html );
+					}
+					?>
+					<div id="close"><a href="#">Close</a></div>
```

Comment:
"Close" needs wrapping in translation function.

Reply:
FYI - text for this has been moved inside the subscribe modal template part. So any translation will need to be handled there.